core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -587,6 +594,30 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
     require(executorTimeoutThresholdMs > executorHeartbeatIntervalMs, "The value of " +
       s"${networkTimeout}=${executorTimeoutThresholdMs}ms must be no less than the value of " +
       s"spark.executor.heartbeatInterval=${executorHeartbeatIntervalMs}ms.")
+
+    // Make sure the executor resources were specified and are large enough if
+    // any task resources were specified.
+    val taskResourcesAndCount =
+    getAllWithPrefixAndSuffix(SPARK_TASK_RESOURCE_PREFIX, SPARK_RESOURCE_COUNT_SUFFIX).toMap
+    val executorResourcesAndCounts =
+      getAllWithPrefixAndSuffix(SPARK_EXECUTOR_RESOURCE_PREFIX, SPARK_RESOURCE_COUNT_SUFFIX).toMap
+
+    taskResourcesAndCount.foreach { case (rName, taskCount) =>
+      val execCount = executorResourcesAndCounts.get(rName).getOrElse(
+        throw new SparkException(
+          s"The executor resource config: " +
+            s"${SPARK_EXECUTOR_RESOURCE_PREFIX + rName + SPARK_RESOURCE_COUNT_SUFFIX} " +
+            "needs to be specified since a task requirement config: " +
+            s"${SPARK_TASK_RESOURCE_PREFIX + rName + SPARK_RESOURCE_COUNT_SUFFIX} was specified")
+      )
+      if (execCount.toLong < taskCount.toLong) {
```

Comment:
Shall we produce a warning if the executor requested count is not a multiple of task count?

Reply:
that should be in the scheduler PR